### PR TITLE
[FIX] web, crm: override of form controller save function

### PR DESCRIPTION
**Pull request by jpp-odoo**
Before this commit, some functions overrode the save function of the form controller. The issue is that this function is only called when the button save is clicked, and not when we save differently (for example, when clicking on the breadcrumb).

To avoid this mistake the save function on the form controller was renamed to: `saveButtonClicked`, and the code that overrode the function now override the save function of the Record, that it's called at each time a save is perform.

### CODE REVIEW

File: addons/crm/static/src/views/fields/crm_statusbar_field.js [hunk 1]
```diff
@@ -2,14 +2,8 @@
```
Comment (Williambraecky) [inline addons/crm/static/src/views/fields/crm_statusbar_field.js:2]: This whole widget is probably not necessary anymore as the code in selectItem is the same

**Review by Williambraecky — APPROVED** [review]
Crm stuff lgtm

**Comment by tivisse** [comment]
cc @awa-odoo @tde-banana-odoo

File: addons/web/static/src/webclient/settings_form_view/settings_form_controller.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ export class SettingsFormController extends formView.Controller {
                     body: message,
                     confirm: async () => {
                         await this.model.root.save({ stayInEdition: true });
-                        await this.save();
+                        await this.saveButtonClicked();
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/settings_form_view/settings_form_controller.js:65]: you are a liar!

**Review by aab-odoo — COMMENTED** [review]
Ok, but I would not rename the function as this is not completely true.

Also, check William's comment :-)

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by jpp-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**